frontend/src/views/Org/SshCaPage/components/SshCertificateModal.tsx

```diff
@@ -0,0 +1,350 @@
+            />
+            {operation === SshCertificateOperation.SIGN_SSH_KEY && (
+              <Controller
+                control={control}
+                name="publicKey"
+                render={({ field, fieldState: { error } }) => (
+                  <FormControl
+                    label="SSH Public Key"
+                    isError={Boolean(error)}
+                    errorText={error?.message}
+                    isRequired
+                  >
+                    <Input {...field} placeholder="ssh-rsa AAA ..." />
+                  </FormControl>
+                )}
+              />
+            )}
+            {operation === SshCertificateOperation.ISSUE_SSH_CREDS && (
+              <Controller
+                control={control}
+                name="keyAlgorithm"
+                defaultValue={CertKeyAlgorithm.RSA_2048}
+                render={({ field: { onChange, ...field }, fieldState: { error } }) => (
+                  <FormControl
+                    label="Key Algorithm"
+                    errorText={error?.message}
+                    isError={Boolean(error)}
+                  >
+                    <Select
+                      defaultValue={field.value}
+                      {...field}
+                      onValueChange={(e) => onChange(e)}
+                      className="w-full"
+                    >
+                      {certKeyAlgorithms.map(({ label, value }) => (
+                        <SelectItem value={String(value || "")} key={label}>
+                          {label}
+                        </SelectItem>
+                      ))}
+                    </Select>
+                  </FormControl>
+                )}
+              />
+            )}
+            <Controller
+              control={control}
+              name="principals"
+              render={({ field, fieldState: { error } }) => (
+                <FormControl
+                  label="Principal(s)"
+                  isError={Boolean(error)}
+                  errorText={error?.message}
+                  isRequired
+                >
+                  <Input {...field} placeholder="ec2-user" />
+                </FormControl>
+              )}
+            />
+
+            <Controller
+              control={control}
+              name="ttl"
+              render={({ field, fieldState: { error } }) => (
+                <FormControl label="TTL" isError={Boolean(error)} errorText={error?.message}>
+                  <Input {...field} placeholder="2 days, 1d, 2h, 1y, ..." />
+                </FormControl>
+              )}
+            />
+            <Controller
+              control={control}
+              name="keyId"
+              render={({ field, fieldState: { error } }) => (
+                <FormControl label="Key ID" isError={Boolean(error)} errorText={error?.message}>
+                  <Input {...field} placeholder="12345678" />
+                </FormControl>
+              )}
+            />
```

Comment:
same for this one here I think. like if custom key ID is not enabled for the template, better if I don't see this at all